### PR TITLE
Simplify ProcessGetBlockData execution by removing send flag

### PR DESCRIPTION
**Pull request by pstratem**
Setting the send flag to false can be replaced by simply returning.

### CODE REVIEW

**Comment by sipa** [comment]
That's the most concise description of what a PR fixes I've ever seen. Can you elaborate on how this will improve @TheBlueMatt?

**Comment by pstratem** [comment]
the send flag effectively exits the function whenever it's set to false, this simply actually returns

**Comment by sipa** [comment]
Yes, I can read the code. But can you actually put that in the PR description? The description goes into the merge commit for example, and is helpful for people investigating the history of code.

File: src/net_processing.cpp [hunk 1]
```diff
@@ -1109,35 +1108,35 @@ void static ProcessGetBlockData(CNode* pfrom, const Consensus::Params& consensus
     LOCK(cs_main);
     const CBlockIndex* pindex = LookupBlockIndex(inv.hash);
     if (pindex) {
```
Comment (skeees) [inline src/net_processing.cpp]: you should return if pindex is not defined

Comment (pstratem) [reply src/net_processing.cpp]: indeed good catch

**Comment by fanquake** [comment]
Travis:
```
11.84s$ if [ "$TRAVIS_EVENT_TYPE" = "pull_request" ]; then contrib/devtools/lint-all.sh; fi
./test/functional/p2p_invalid_getdata.py:31:1: W293 blank line contains whitespace
^---- failure generated from contrib/devtools/lint-python.sh
```

File: test/functional/p2p_invalid_getdata.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
+# Copyright (c) 2016-2017 The Bitcoin Core developers
```
Comment (Empact) [inline test/functional/p2p_invalid_getdata.py]: 2018

File: test/functional/p2p_invalid_getdata.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+# TestP2PConn: A peer we use to send messages to bitcoind, and store responses.
+class TestP2PConn(P2PInterface):
+    def __init__(self):
+        super().__init__()
+
+class InvalidGetDataTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 3
```
Comment (Empact) [inline test/functional/p2p_invalid_getdata.py]: Looks like you're using 2 nodes below, index 0 and 1, is that right?

File: test/functional/p2p_invalid_getdata.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+class InvalidGetDataTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+
+    def run_test(self):
+        # Setup the p2p connections and start up the network thread.
+        self.test_node = self.nodes[0].add_p2p_connection(TestP2PConn())
+
+        network_thread_start()
```
Comment (MarcoFalke) [inline test/functional/p2p_invalid_getdata.py]: nit: The thread is always running on master. Remove this line.

**Comment by fanquake** [comment]
I've rebased and fixed the nit in #13670.

**Pull request closed without merging**